### PR TITLE
gsheetsImport: reduce the number of API calls

### PR DESCRIPTION
**Pull request by alanb-sony**
Uploading test results from multiple platforms can hit the google API rate limits. This patch reduces the number of reads and writes made so should reduce the chance of hitting the limits.

### CODE REVIEW

File: utilities/run-test-suites/gsheetsImport/resultsImporter.py [hunk 1]
```diff
@@ -31,6 +32,26 @@
                "Not Applicable"]
 
 
+def get_range(worksheet_data, start_row, start_col, end_row, end_col):
```
Comment (garethsb-sony) [inline utilities/run-test-suites/gsheetsImport/resultsImporter.py:35]: `get_range` is definitely better than the string hackery and `gspread.utils.range` calls we had before

File: utilities/run-test-suites/gsheetsImport/resultsImporter.py [hunk 2]
```diff
@@ -128,14 +146,15 @@ def gsheets_import(test_results, worksheet, filename, start_col=1):
...
+    if not ranges_equal(original_cell_list_names, cell_list_names):
+        worksheet.update_cells(cell_list_names)
     # 'USER_ENTERED' allows formulae to be used
     worksheet.update_cells(cell_list_results, value_input_option='USER_ENTERED')
 
 
 def main():
     parser = argparse.ArgumentParser()
-    parser.add_argument("--json", required=True)
+    parser.add_argument("--json", required=True, action="append")
```
Comment (garethsb-sony) [inline utilities/run-test-suites/gsheetsImport/resultsImporter.py:157]: Allowing multiple files via `--json` seems useful

File: utilities/run-test-suites/gsheetsImport/resultsImporter.py [hunk 3]
```diff
@@ -31,6 +32,26 @@
...
+    for i in range(len(first)):
+        if (first[i].row != second[i].row or
+                first[i].col != second[i].col or
+                first[i].value != second[i].value):
+            return False
+    return True
+
+
 def gsheets_import(test_results, worksheet, filename, start_col=1):
     """Upload results data to spreadsheet"""
```
Comment (garethsb-sony) [inline utilities/run-test-suites/gsheetsImport/resultsImporter.py:57]: I wonder if we should be using `worksheet_get_all_values(ValueRenderOption.FORMULA)` given we're using `'USER_ENTERED'` later?

**Comment by garethsb-sony** [comment]
Going to look at using `append_rows` (or `insert_row(s)`, after concerns mentioned in https://github.com/burnash/gspread/pull/734 have been addressed to make that atomic) to reduce API usage further. Using `insert_row(s)` would enable us to add a command-line flag to insert at the top so in large spreadsheets of historical results new stuff is easy to find.

**Comment by burnash** [comment]
@alanb-sony would you be interested in contributing your atomic version of insert_rows() to gspread?

**Comment by alanb-sony** [comment]
@burnash yes if you think the implementation is reasonable?

**Comment by burnash** [comment]
I think so. Would it be possible to contribute a multi-rows version of the implementation in this PR?
gspread already has `insert_row()`. I've merged https://github.com/burnash/gspread/pull/734 with `insert_rows()` but haven't released it yet. I believe if there's a chance to reduce the number of API calls we should certainly use it.

**Pull request merged**